src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/client/DeliusClient.kt

```diff
@@ -41,7 +42,14 @@ class DeliusClient(
     parameters = mapOf("page" to listOf(page), "size" to listOf(pageSize)),
   ).body!!
 
-  fun findByCrn(crn: String) = get<PersonalDetailsOverview>("/case-summary/$crn") { Mono.empty() }?.body
+  fun findByCrn(crn: String): PersonalDetailsOverview? {
+    val response = try {
+      get<PersonalDetailsOverview>("/case-summary/$crn") { Mono.empty() }?.body
```

Comment:
The lambda being passed in meant to deal with the NOT_FOUND status case. Do you know why it isn't working in this case? Or am I misunderstanding the problem?

Reply:
As best I could determine, between the behaviour being shown and rummaging in this client source code:
This is because it trying to first cast the body to a PersonalDetailsOverview which fails because the body we get with a 404 has the fields `status` and `message` so it's never getting to the handler as an exception is being thrown.

Reply:
For additional reference: I found this being handled in a similar way in the Risk Service (https://github.com/ministryofjustice/make-recall-decision-api/blob/ea7f48c4ff1161acf1e435a653b77ce1c2a32b3e/src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/risk/RiskService.kt#L167) from which I took cues.

Reply:
I wasn't expecting it to try to cast in that case, but maybe I've missed how some of this works. Let's get this fix in and can look into the problem further if needed.